python/interpret_community/mimic/mimic_explainer.py

```diff
@@ -239,9 +239,12 @@ def __init__(self, model, initialization_examples, explainable_model, explainabl
         self.reset_index = reset_index
         self._datamapper = None
         if transformations is not None:
-            self._datamapper, initialization_examples = get_datamapper_and_transformed_data(
+            self._datamapper, new_initialization_examples = get_datamapper_and_transformed_data(
                 examples=initialization_examples, transformations=transformations,
                 allow_all_transformations=allow_all_transformations)
+            # memory optimization to clear all internal references so they can be gc'ed as soon as possible
+            initialization_examples._clear()
+            initialization_examples = new_initialization_examples
         if reset_index != ResetIndex.Ignore:
```

Comment:
Shouldn't python garbage collect if initialization_examples is not longer in use. Why do we need to clear this here?

Reply:
yes, it is garbage collected after designer wrapper loses reference to passed initialization_examples, but we can free the memory much, much earlier by clearing the references after we are done here, thereby reducing overall memory usage in that scenario by 2 GB